front/src/features/filter/TimeDropdown.tsx

```diff
@@ -1,23 +1,43 @@
 import { Dropdown } from "@/components/molecules/Dropdown";
 import { useNavigation } from "@/hooks/useNavigation";
 import { TimeOption } from "@/types/logDisplayOptions";
+import { DatePickerPopup } from "./DatePickerPopup";
+import { useState } from "react";
 
 export const TimeDropdown = () => {
     const { searchParams, updateSearchParam } = useNavigation();
-    const timeOptions = ["Last hour", "Last day", "Last week"];
-    let selectedTime = searchParams.get("dateTimeFilter");
-    if (!selectedTime || !timeOptions.includes(selectedTime)) {
-        selectedTime = "Last hour";
+    const [isPopupOpen, setIsPopupOpen] = useState(false);
+    const [selectedTime, setSelectedTime] = useState<string | null>(searchParams.get("dateTimeFilter"));
+
+    const timeOptions = ["Last hour", "Last day", "Last week", "Filter by time"];
+    if (selectedTime == null || !timeOptions.includes(selectedTime)) {
+        setSelectedTime("Last hour");
     }
-    const onTimeSelect = (newValue: string) => {
-        if (timeOptions.includes(newValue))
+
+    const onSelectTime = (newValue: string) => {
+        setSelectedTime(newValue);
+        if (newValue !== "Filter by time") {
             updateSearchParam({ dateTimeFilter: newValue as TimeOption });
+        } else {
+            setIsPopupOpen(true);
+        }
     };
+
+    const cancelCustomRangePicker = () => {
+        setIsPopupOpen(false);
+    };
+
     return (
-        <Dropdown
-            options={timeOptions}
-            selected={selectedTime}
-            onSelect={onTimeSelect}
-        />
+        <>
+            <Dropdown
+                options={timeOptions}
+                selected={selectedTime ?? "Last hour"}
```

Comment:
Why was the `selectedTime ?? "Last hour"` added?

Reply:
This is because selectedTime can be null and selected does not accept null values